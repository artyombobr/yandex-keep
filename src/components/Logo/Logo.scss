.logo {
  display: flex;
  align-items: center;

  &__item {
    display: inline-block;
    margin-right: 5px;
  }
}
