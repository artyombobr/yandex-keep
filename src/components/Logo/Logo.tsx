import React from 'react';
import './Logo.scss';

interface LogoProps {
  className: string;
  color?: string;
}

function Logo(props: LogoProps) {
  const { className, color } = props;
  return (
    <div className={`${className} logo`}>
      <a className="logo__item" href="https://yandex.ru">
        <img
          src={`https://yastatic.net/q/logoaas/v1/Яндекс.svg?color=${color}`}
          alt="Яндекс"
          style={{ width: 72 }}
        />
      </a>
      <a className="logo__item" href="/">
        <img
          src={`https://yastatic.net/q/logoaas/v1/Домашка.svg?color=${color}`}
          alt="Композитор"
          style={{ width: 94 }}
        />
      </a>
    </div>
  );
}

export default Logo;
