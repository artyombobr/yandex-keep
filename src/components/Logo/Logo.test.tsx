import React from 'react';
import { shallow } from 'enzyme';
import Logo from './Logo';

describe('Logo', () => {
  it('should match snapshot', () => {
    const component = shallow(<Logo className="header__logo" color="black" />);
    expect(component).toMatchSnapshot();
  });
});
