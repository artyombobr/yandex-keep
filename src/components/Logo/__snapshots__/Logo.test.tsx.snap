// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logo
    className="header__logo"
    color="black"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="logo__item"
          href="https://yandex.ru"
        >
          <img
            alt="Яндекс"
            src="https://yastatic.net/q/logoaas/v1/Яндекс.svg?color=black"
            style={
              Object {
                "width": 72,
              }
            }
          />
        </a>,
        <a
          className="logo__item"
          href="/"
        >
          <img
            alt="Композитор"
            src="https://yastatic.net/q/logoaas/v1/Домашка.svg?color=black"
            style={
              Object {
                "width": 94,
              }
            }
          />
        </a>,
      ],
      "className": "header__logo logo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Яндекс"
            src="https://yastatic.net/q/logoaas/v1/Яндекс.svg?color=black"
            style={
              Object {
                "width": 72,
              }
            }
          />,
          "className": "logo__item",
          "href": "https://yandex.ru",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Яндекс",
            "src": "https://yastatic.net/q/logoaas/v1/Яндекс.svg?color=black",
            "style": Object {
              "width": 72,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Композитор"
            src="https://yastatic.net/q/logoaas/v1/Домашка.svg?color=black"
            style={
              Object {
                "width": 94,
              }
            }
          />,
          "className": "logo__item",
          "href": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Композитор",
            "src": "https://yastatic.net/q/logoaas/v1/Домашка.svg?color=black",
            "style": Object {
              "width": 94,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="logo__item"
            href="https://yandex.ru"
          >
            <img
              alt="Яндекс"
              src="https://yastatic.net/q/logoaas/v1/Яндекс.svg?color=black"
              style={
                Object {
                  "width": 72,
                }
              }
            />
          </a>,
          <a
            className="logo__item"
            href="/"
          >
            <img
              alt="Композитор"
              src="https://yastatic.net/q/logoaas/v1/Домашка.svg?color=black"
              style={
                Object {
                  "width": 94,
                }
              }
            />
          </a>,
        ],
        "className": "header__logo logo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Яндекс"
              src="https://yastatic.net/q/logoaas/v1/Яндекс.svg?color=black"
              style={
                Object {
                  "width": 72,
                }
              }
            />,
            "className": "logo__item",
            "href": "https://yandex.ru",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Яндекс",
              "src": "https://yastatic.net/q/logoaas/v1/Яндекс.svg?color=black",
              "style": Object {
                "width": 72,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Композитор"
              src="https://yastatic.net/q/logoaas/v1/Домашка.svg?color=black"
              style={
                Object {
                  "width": 94,
                }
              }
            />,
            "className": "logo__item",
            "href": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Композитор",
              "src": "https://yastatic.net/q/logoaas/v1/Домашка.svg?color=black",
              "style": Object {
                "width": 94,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
